analyzers/src/SonarAnalyzer.VisualBasic/Rules/ArrayPassedAsParams.cs

```diff
@@ -0,0 +1,48 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.VisualBasic;
+
+[DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+public sealed class ArrayPassedAsParams : ArrayPassedAsParamsBase<SyntaxKind, ArgumentSyntax>
+{
+    protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
+
+    protected override SyntaxKind[] ExpressionKinds { get; } =
+        {
+            SyntaxKind.ObjectCreationExpression,
+            SyntaxKind.InvocationExpression
+        };
+
+    protected override ArgumentSyntax GetLastArgumentIfArrayCreation(SyntaxNode expression) =>
+        expression switch
+        {
+            ObjectCreationExpressionSyntax { } creation => GetLastArgumentIfArrayCreation(creation.ArgumentList),
+            InvocationExpressionSyntax { } invocation => GetLastArgumentIfArrayCreation(invocation.ArgumentList),
+            _ => null
+        };
+
+    private static ArgumentSyntax GetLastArgumentIfArrayCreation(ArgumentListSyntax argumentList) =>
+        argumentList is { Arguments: { Count: > 0 } arguments }
+        && arguments.Last().GetExpression() is ArrayCreationExpressionSyntax invocationArray
+        && invocationArray.Initializer is CollectionInitializerSyntax { Initializers.Count: > 0 }
+            ? arguments.Last()
+            : null;
```

Comment:
Slight improvement and more in line with the C# version:
```suggestion
        && arguments.Last() is var argument
        && argument.GetExpression() is ArrayCreationExpressionSyntax { Initializer: CollectionInitializerSyntax { Initializers.Count: > 0 } }
            ? argument
            : null;
```